### ISSUE TITLE
Filtering of `eachrow(df)` not working in 1.4.0

### ISSUE THREAD

**Issue by dalum** [JuliaData/DataFrames.jl#3191]
On 1.3.6:
```
julia> df = DataFrame(:a => [1, 2, 3, 4])
4×1 DataFrame
 Row │ a     
     │ Int64 
─────┼───────
   1 │     1
   2 │     2
   3 │     3
   4 │     4

julia> rows = filter(row -> row[:a] > 2, eachrow(df))
2-element Vector{DataFrameRow}:
 DataFrameRow
 Row │ a     
     │ Int64 
─────┼───────
   3 │     3
 DataFrameRow
 Row │ a     
     │ Int64 
─────┼───────
   4 │     4
```

On 1.4.0:
```
julia> df = DataFrame(:a => [1, 2, 3, 4])
rows 4×1 DataFrame
 Row │ a     
     │ Int64 
─────┼───────
   1 │     1
   2 │     2
   3 │     3
   4 │     4

julia> rows = filter(row -> row[:a] > 2, eachrow(df))
0×1 DataFrameRows
```
I couldn't see anything in the release notes about this. Is this new behaviour expected?

**Comment by bkamins** [comment]
This is a bug.

**Comment by bkamins** [comment]
The change is mentioned in `Performance` section of NEWS.md:

Make one-dimensional multi-element indexing of DataFrameRows return DataFrameRows (https://github.com/JuliaData/DataFrames.jl/pull/3037)

It is fixed in #3192. The bug is that `getindex` aliased passed selector vector in some cases (and `filter` in Base for some reason later empties this vector).

The change I describe is that now:
```
julia> df = DataFrame(:a => [1, 2, 3, 4])
4×1 DataFrame
 Row │ a
     │ Int64
─────┼───────
   1 │     1
   2 │     2
   3 │     3
   4 │     4

julia> eachrow(df)[2:3]
2×1 DataFrameRows
 Row │ a
     │ Int64
─────┼───────
   1 │     2
   2 │     3
```
returns `DataFrameRows` (not a `Vector` of `DataFrameRow`) as this will be significantly faster.
